src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -517,6 +472,61 @@ bool CascadiaSettings::_isPowerShellCoreInstalledInPath(const std::wstring_view
     return false;
 }
 
+// Function Description:
+// - Adds all of the WSL profiles to the provided container.
+// Arguments:
+// - A ref to the profiles container where the WSL profiles are to be added
+// Return Value:
+// - <none>
+void CascadiaSettings::_CreateWslProfiles(std::vector<TerminalApp::Profile>& profileStorage)
+{
+    try
+    {
+        DWORD bytesAvailable = 0;
+        DWORD exitCode = 0;
+        std::wstring command = L"wsl.exe --list";
+        std::wstring wline = L"";
+        std::wstring distName = L"";
+        wil::unique_handle readPipe;
+        wil::unique_handle writePipe;
+        SECURITY_ATTRIBUTES sa{ sizeof(sa), nullptr, true };
+        THROW_IF_WIN32_BOOL_FALSE(CreatePipe(&readPipe, &writePipe, &sa, 0));
+        STARTUPINFO si{};
+        si.cb = sizeof(si);
+        si.dwFlags = STARTF_USESTDHANDLES;
+        si.hStdOutput = writePipe.get();
+        si.hStdError = writePipe.get();
+        wil::unique_process_information pi{};
+
+        THROW_IF_WIN32_BOOL_FALSE(CreateProcessW(nullptr, const_cast<LPWSTR>(command.c_str()), nullptr, nullptr,
+                                                    TRUE, CREATE_NO_WINDOW, nullptr, nullptr, &si, &pi));
+        if (WaitForSingleObject(pi.hProcess, INFINITE) != NULL)
+        {
+            throw;

```

Comment:
Should probably throw *something* in particular like with `THROW_HR()`

Reply:
I went through part of the list (damn, there's a lot of these!)  I narrowed it down to the following:

```
ERROR_INVALID_DATA
ERROR_READ_FAULT
ERROR_SEM_TIMEOUT
ERROR_WAIT_NO_CHILDREN
ERROR_CHILD_NOT_COMPLETE
ERROR_UNHANDLED_EXCEPTION
```

and went with these:

```
ERROR_CHILD_NOT_COMPLETE
ERROR_UNHANDLED_EXCEPTION
```

Reply:
Yeah that works.